src/d3a/models/area/markets.py

```diff
@@ -15,84 +15,158 @@
 You should have received a copy of the GNU General Public License
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
 """
-from pendulum import DateTime # noqa
-from typing import Dict  # noqa
-
-from d3a.models.market import GridFee
-from d3a.models.market.two_sided import TwoSidedMarket
-from d3a.models.market.one_sided import OneSidedMarket
-from d3a.models.market.balancing import BalancingMarket
-from d3a.models.market import Market # noqa
-from d3a_interface.constants_limits import ConstSettings
 from collections import OrderedDict
-from d3a.d3a_core.util import is_timeslot_in_simulation_duration
+from typing import Dict, List, TYPE_CHECKING
+
 import d3a.constants
+from d3a.d3a_core.util import is_timeslot_in_simulation_duration
+from d3a.models.market import GridFee, Market
+from d3a.models.market.balancing import BalancingMarket
+from d3a.models.market.one_sided import OneSidedMarket
+from d3a.models.market.two_sided import TwoSidedMarket
+from d3a_interface.constants_limits import ConstSettings, GlobalConfig
+from pendulum import DateTime
+
+if TYPE_CHECKING:
+    from d3a.models.area import Area
 
 
 class AreaMarkets:
+    """Class that holds all markets for areas."""
+
     def __init__(self, area_log):
-        # Children trade in `markets`
         self.log = area_log
+        # Children trade in `markets`
         self.markets = OrderedDict()  # type: Dict[DateTime, Market]
         self.balancing_markets = OrderedDict()  # type: Dict[DateTime, BalancingMarket]
         # Past markets
         self.past_markets = OrderedDict()  # type: Dict[DateTime, Market]
         self.past_balancing_markets = OrderedDict()  # type: Dict[DateTime, BalancingMarket]
-        self._indexed_future_markets = {}
+        # Settlement markets
+        self.settlement_markets = OrderedDict()  # type: Dict[DateTime, Market]
+        self.past_settlement_markets = OrderedDict()  # type: Dict[DateTime, Market]
+        self.indexed_future_markets = {}
 
     @property
-    def indexed_future_markets(self):
-        return self._indexed_future_markets
+    def all_future_spot_markets(self) -> List:
+        """Return all future markets in a list."""
+        return list(self.markets.values())
 
-    @property
-    def all_spot_markets(self):
-        return list(self.markets.values()) + list(self.past_markets.values())
+    def _update_indexed_future_markets(self) -> None:
+        """Update the indexed_future_markets mapping."""
+        self.indexed_future_markets = {m.id: m for m in self.all_future_spot_markets}
```

Comment:
I would postpone this discussion until we implement the new behaviour of the futures market. For now I think the time_slot dependant buffering of the markets, balancing markets and settlement markets is more handy. 
But maybe you have a better argument to change this to ids? How does anybody else think about it?

Reply:
Timeslot indexing is necessary IMO, and much more helpful than id-indexing. The only reason that it exists here is for performance improvement, because the operation of searching a market with an id is quite frequent therefore there was a significant performance gain from adding the indexed version. I agree with Hannes though that once we add the futures market this will become obsolete, therefore we should remove it then

Reply:
I would say that the important thing is to keep these markets variables consistent with each other. If we do so, we'll be able to unify their implementation in the future much more easily.